x-pack/plugins/maps/public/routes/list_page/maps_list_view.tsx

```diff
@@ -119,19 +121,35 @@ async function deleteMaps(items: object[]) {
   await Promise.all(deletions);
 }
 
-export function MapsListView() {
+export function MapsListView({ stateTransfer }: { stateTransfer: EmbeddableStateTransfer }) {
+  useEffect(
+    () => {
+      stateTransfer.clearEditorState(APP_ID);
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    []
+  );
+  const [showEmptyList, setShowEmptyList] = useState(false);
   const isReadOnly = !getMapsCapabilities().save;
 
   getCoreChrome().docTitle.change(getAppTitle());
   getCoreChrome().setBreadcrumbs([{ text: getAppTitle() }]);
 
   return (
     <TableListView
+      noItemsFragment={showEmptyList ? undefined : <Redirect to="/map" />}
```

Comment:
Why not just get ride of the re-direct all together? I think @thomasneirynck comments on keeping it were to not muck with load_list_and_render.tsx, but since you are removing that in this PR, just go for the full thing and remove the redirect altogehter. That will get ride of setShowEmptyList, which is hard to follow why it even exists without a bit of history for the component.

If you want to keep the redirect, then keep load_list_and_render.tsx and get ride of setShowEmptyList. Really the fix is just the changes in table_list_view.tsx

Reply:
There was some offline discussion on this where I proposed to just remove the redirect altogether since really that would be easier. @thomasneirynck was not in favor of that at this point. This PR preserves the existing behavior but shows the empty list on delete